### PR TITLE
fix: make the bi-directional encryptor configurable in EncryptionUtils

### PR DESCRIPTION
**Pull request by jkevan**
### Description

This pull request introduces a configurable encryption mechanism to the `EncryptionUtils` class, allowing the encryption password and algorithm to be set via environment variables, system properties, or programmatically. The changes ensure backward compatibility, provide explicit initialization methods, and add comprehensive unit tests to verify different configuration scenarios and edge cases.

### Encryption configuration and initialization

* Added support for configuring the encryption password and algorithm through environment variables (`JAHIA_COMMONS_ENCRYPTOR_PASSWORD`, `JAHIA_COMMONS_ENCRYPTOR_ALGORITHM`), system properties (`jahia-commons.encryptor.password`, `jahia-commons.encryptor.algorithm`), or programmatically via new initialization methods in `EncryptionUtils`. This allows flexible and secure configuration for different deployment environments. [[1]](diffhunk://#diff-fd9a62a1c295568bfe9e4e55bba14c437ccea2fe16d888b92b3049429ef2f025R59-R72) [[2]](diffhunk://#diff-fd9a62a1c295568bfe9e4e55bba14c437ccea2fe16d888b92b3049429ef2f025R154-R219)
* Introduced `EncryptionUtils.initializeEncryptor(...)` methods to explicitly initialize the encryptor before use, with options to set password, algorithm, and optionally force reinitialization (mainly for testing). Attempting to reinitialize without force throws an exception to prevent accidental misconfiguration.
* Refactored lazy initialization logic for the encryptor to ensure thread safety and proper configuration precedence (environment variable > system property > default).

### Utility and legacy support

* Added new `ConfigurationUtils` class to centralize configuration value retrieval from environment variables, system properties, and defaults.
* Marked the legacy SHA-1 digest method as deprecated and updated related documentation.

### Testing

* Added comprehensive unit tests in `EncryptionUtilsTest` to cover default configuration, custom password/algorithm initialization, reinitialization behavior, and configuration via system properties. Tests also verify encryption consistency and backward compatibility.

These changes significantly improve the flexibility, security, and testability of encryption configuration in the codebase.

### CODE REVIEW

File: src/main/java/org/jahia/commons/encryption/EncryptionUtils.java [hunk 1]
```diff
@@ -136,7 +138,9 @@ public static String pbkdf2Digest(String source, boolean prefixWithId) {
      * @param source
      *            the source text to be digested
      * @return the Base64 encoded SHA-1 digest of the provided text
+     * @deprecated in Jahia 7 a more robust PBKDF2 algorithm is used for password hashing. The previous SHA-1 based algorithm is no longer used
      */
+    @Deprecated
```
Comment (baptistegrimaud) [inline src/main/java/org/jahia/commons/encryption/EncryptionUtils.java:143]: should we add a `since` parameter here?

File: src/main/java/org/jahia/commons/encryption/ConfigurationUtils.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
+/*
+ * ==========================================================================================
+ * =                   JAHIA'S DUAL LICENSING - IMPORTANT INFORMATION                       =
+ * ==========================================================================================
+ *
+ *                                 http://www.jahia.com
+ *
+ *     Copyright (C) 2002-2019 Jahia Solutions Group SA. All rights reserved.
```
Comment (baptistegrimaud) [inline src/main/java/org/jahia/commons/encryption/ConfigurationUtils.java:8]: Never know if that's important or not 🤷‍♂️ :
```suggestion
 *     Copyright (C) 2002-2025 Jahia Solutions Group SA. All rights reserved.
```

File: src/main/java/org/jahia/commons/encryption/ConfigurationUtils.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+ *
+ *     If you are unsure which license is appropriate for your use,
+ *     please contact the sales department at sales@jahia.com.
+ */
+package org.jahia.commons.encryption;
+
+/**
+ * Configuration utilities for retrieving values from environment variables and system properties.
+ *
+ * @author Jahia Solutions Group SA
```
Comment (baptistegrimaud) [inline src/main/java/org/jahia/commons/encryption/ConfigurationUtils.java:49]: Minor: do we actually use those `@author` tags?

File: src/main/java/org/jahia/commons/encryption/EncryptionUtils.java [hunk 4]
```diff
@@ -147,4 +151,70 @@ public static String sha1DigestLegacy(String source) {
...
+     */
+    public static void initializeEncryptor(String password, String algorithm) {
+        initializeEncryptor(password, algorithm, false);
+    }
+
+    /**
+     * Allows applications to initialize the encryptor configuration before first use.
+     * This method should be called during application startup, before any encryption operations.
+     *
+     * <p><strong>WARNING:</strong> Using the force parameter to reinitialize an encryptor that has
```
Comment (baptistegrimaud) [inline src/main/java/org/jahia/commons/encryption/EncryptionUtils.java:171]: Should we mention somewhere that setting the env variables / Jahia properties after any encryption operation is performed will make the old data undecryptable? Or maybe that will be done on an Academy page?

**Review by baptistegrimaud — APPROVED** [review]

File: src/main/java/org/jahia/commons/encryption/EncryptionUtils.java (hunk 1 above)
Comment (jkevan) [reply src/main/java/org/jahia/commons/encryption/EncryptionUtils.java:143]: Not interesting in library that small I would say.
